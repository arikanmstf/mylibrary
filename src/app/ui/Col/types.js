/**
 * Component Types By => create-module script
 * @version 1.0.0
 *
 */

// @flow
export type ColProps = {
  style?: ?Object,
};
