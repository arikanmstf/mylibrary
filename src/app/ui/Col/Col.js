/**
 * Web Component Template By => create-module script
 * @version 1.0.0
 *
 */

// @flow
import React from 'react';

import type { Node } from 'react';
import type { ColProps } from './types';
import defaultStyle from './style.json';

const Col = (props: ColProps): Node => {
  const { style, ...other } = props;
  const mergedStyle = {
    ...defaultStyle,
    ...style,
  };

  return (
    <div
      style={mergedStyle}
      {...other}
    />
  );
};

export default Col;
