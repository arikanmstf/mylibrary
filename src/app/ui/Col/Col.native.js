/**
 * Native Component Template By => create-module script
 * @version 1.0.0
 *
 */

// @flow
import React from 'react';
import { StyleSheet, View } from 'react-native';

import type { Node } from 'react';
import type { ColProps } from './types';
import defaultStyle from './style.json';

const Col = (props: ColProps): ?Node => {
  const { style, ...other } = props;
  const mergedStyles = StyleSheet.create({
    container: {
      ...defaultStyle,
      ...style,
    },
  });

  return (
    <View
      style={mergedStyles.container}
      {...other}
    />
  );
};

export default Col;
