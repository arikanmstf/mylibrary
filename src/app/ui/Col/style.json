{
  "width": "90%",
  "padding": 10
}
