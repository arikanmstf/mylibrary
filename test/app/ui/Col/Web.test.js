/**
 * Web Component Test Template By => create-module script
 * @version 1.0.0
 *
 */

import React from 'react';
import { shallow } from 'enzyme';
import Col from 'ui/Col/Web';

describe('test/app/ui/Col/Web.test.js', () => {
  it('Render', () => {
    const props = {};
    const wrapper = shallow(<Col {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
