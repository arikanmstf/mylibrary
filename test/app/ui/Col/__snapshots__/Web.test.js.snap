// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test/app/ui/Col/Web.test.js Render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Col />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "style": Object {
        "flexDirection": "column",
        "justifyContent": "center",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "style": Object {
          "flexDirection": "column",
          "justifyContent": "center",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
